longhorn/longhorn-tests
# test: add Dockerfile and test.yaml to build negative testing image and run it as a pod

test: add Dockerfile and test.yaml to build negative testing image and run it as a pod

Provide 2 ways to run negative testing
(1) out of cluster
(2) in cluster

Steps to run negative testing out of cluster:
(1) make sure `KUBECONFIG` is correct, and `kubectl` can access the cluster
(2) expose Longhorn url using node port or port forward
(3) export `LONGHORN_CLIENT_URL` as environment variable
(4) `cd e2e/` and run the test `./run.sh -t "Replica Rebuilding While Replica Deletion"`

Steps to run negative testing in cluster:
(1) build test image: `docker build -t test-e2e -f e2e/Dockerfile .` from longhorn-tests directory
(2) modify the content of `test.yaml` to your desired values, and execute `kubectl apply -f e2e/deploy/test.yaml`
(3) view logs: `kubectl logs longhorn-test -c longhorn-test -f`
(4) to get the `robot` test report, currently need to manually copy them from the test pod:
```
kubectl cp longhorn-test:/tmp/test-report/report.html report.html -c longhorn-test-report
kubectl cp longhorn-test:/tmp/test-report/log.html log.html -c longhorn-test-report
```

For https://github.com/longhorn/longhorn/issues/6502

Signed-off-by: Yang Chiu <yang.chiu@suse.com>
